pages/k8s/logging.md

```diff
@@ -156,7 +162,111 @@ the following:
 - Setting up remote logging
 - More advanced filtering and additional examples
 
+<a id="graylog"> </a>
+
+## Cluster logs with Graylog
+
+The recommended way to retrive logs from your cluster is to use a combination
+of **Elasticsearch**, **Graylog** and **Filebeat**. These provide a dashboard
+from which you can monitor both machine-level and cluster-level logs.
+See the [quickstart guide][quickstart] for more details on installing **Charmed Kubernetes**.
+
+### Installation
+
+You can install **Charmed Kubernetes** with Graylog logging using the Juju
+bundle along with the following overlay file
+([download it here][logging-egf-overlay]):
+
+```yaml
+applications:
+  apache2:
+    charm: cs:bionic/apache2
+    num_units: 1
+    expose: true
+    options:
+      enable_modules: "headers proxy_html proxy_http"
+  elasticsearch:
+    charm: cs:bionic/elasticsearch
+    constraints: mem=7G root-disk=16G
+    num_units: 1
+    options:
+      apt-repository: "deb https://artifacts.elastic.co/packages/6.x/apt stable main"
+  filebeat:
+    charm: cs:bionic/filebeat
+    options:
+      install_sources: "deb https://artifacts.elastic.co/packages/6.x/apt stable main"
+      kube_logs: True
+  graylog:
+    charm: cs:bionic/graylog
+    constraints: mem=7G root-disk=16G
+    num_units: 1
+    options:
+      channel: "3/stable"
+  mongodb:
+    charm: cs:bionic/mongodb
+    num_units: 1
+    options:
+      extra_daemon_options: "--bind_ip_all"
+relations:
+  - ["apache2:reverseproxy", "graylog:website"]
+  - ["graylog:elasticsearch", "elasticsearch:client"]
+  - ["graylog:mongodb", "mongodb:database"]
+  - ["filebeat:beats-host", "kubernetes-master:juju-info"]
+  - ["filebeat:beats-host", "kubernetes-worker:juju-info"]
+  - ["filebeat:logstash", "graylog:beats"]
+```
+
+To use this overlay with the **Charmed Kubernetes** bundle, specify it
+during deploy like this:
+
+```bash
+juju deploy charmed-kubernetes --overlay ~/path/logging-egf-overlay.yaml
+```
+
+If you wish to add Graylog logging to an existing deployment, you can export a
+bundle of your current environment and then redeploy it on top of itself with
+the overlay:
+
+```bash
+juju export-bundle --filename mybundle.yaml
+juju deploy ./mybundle.yaml --overlay ~/path/logging-egf-overlay.yaml
+```
+
+At this point, all the applications can communicate with each other, but we
+need to configure the reverse proxy to access the Graylog web interface:
+
+```bash
+juju config apache2 vhost_http_template="$(base64 <vhost-tmpl>)"
+```
+
+A sample reverse proxy template can be found at https://raw.githubusercontent.com/conjure-up/spells/master/charmed-kubernetes/addons/graylog/steps/01_install-graylog/graylog-vhost.tmpl.
```

Comment:
@evilnick this should link to the template in `./assets/graylog-vhost.tmpl`. I wasn't sure how to do a relative link in the docs.  It's probably as simple as `href=./assets/blah`, but I defer to your guru status to make this right.

Reply:
No this is right - www.ubuntu.com doesn't have any of our assets, we fetch them from gh.